packages/contracts/test/FreeForAllExcubia.test.ts

```diff
@@ -0,0 +1,162 @@
+import { expect } from "chai"
```

Comment:
do you think we should keep tests both in hardhat and in foundry? imo could probably just do it in Solidity

Reply:
They are important when you want to test things in js/ts for usage on the web. Foundry helps you to hardcore test your contracts in Solidity but can miss few nuances from js/ts testing on web-side (e.g., big integers params, ...). 

See https://metamask.io/news/developers/hardhat-vs-foundry-choosing-the-right-ethereum-development-tool/

Reply:
Smart